storage/src/buffer.rs

```diff
@@ -0,0 +1,332 @@
+extern crate memmap;
+extern crate omap;
+
+use self::memmap::Mmap;
+use self::omap::OrderedHashMap;
+use std::path::PathBuf;
+use std::fs::{self, File};
+use std::sync::{Arc, Condvar, Mutex, MutexGuard, RwLock, RwLockWriteGuard};
+use std::mem;
+use std::cmp::{max, min};
+use std::collections::HashSet;
+use std::ops::Deref;
+
+const DEFAULT_CAPACITY: usize = 1000;
+const TEMP_RECORD_PREFIX: char = '$';
+
+pub struct Value {
```

Comment:
Wondering if we should use a name other than `Value` since we're already using `Value` in the type system?